src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -650,9 +648,9 @@ private async Task<ConnectBlocksResult> ConnectChainAsync(ChainedHeader newTip,
 
                 await this.ResyncPeersAsync(peersToResync).ConfigureAwait(false);
 
-                if (this.network.Consensus.MaxReorgLength != 0)
+                if (this.chainState.MaxReorgLength != 0)
```

Comment:
Whats the difference?

Reply:
The main issue was that we set the MaxReorgLength on ChainState in the test but we check the `MaxReorgLength`  on network when we want to save it, so `FinalizedBlock` doesn't get saved.

We CAN however still check network's `MaxReorgLength` in CM but then we need to set network's MaxReorgLength in the test (this might be an issue as network's MaxReorgLength  currently has a protected set).

Reply:
In this case try to use reflection?